tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_graphs_reducers_test.ts

```diff
@@ -0,0 +1,463 @@
+        graphs: createDebuggerGraphsState({
+          focusedOp: {
+            graphId: 'g1',
+            opName: 'TestOp_1',
+          },
+        }),
+      });
+      const nextState = reducers(
+        state,
+        actions.graphOpFocused({graph_id: 'g2', op_name: 'TestOp_12'})
+      );
+      expect(nextState.graphs.focusedOp).toEqual({
+        graphId: 'g2',
+        opName: 'TestOp_12',
+      });
+    });
+  });
+
+  describe('graphOpInfoRequested', () => {
+    it('creates key for new graph_id', () => {
+      const state = createDebuggerState();
+      const nextState = reducers(
+        state,
+        actions.graphOpInfoRequested({graph_id: 'g8', op_name: 'x'})
+      );
+      expect(nextState.graphs.loadingOps).toEqual({
+        g8: {x: DataLoadState.LOADING},
+      });
+    });
+
+    it('adds op to existing graph key', () => {
+      const state = createDebuggerState({
+        graphs: createDebuggerGraphsState({
+          loadingOps: {
+            g1: {Op1: DataLoadState.LOADING},
+            g2: {Op2: DataLoadState.LOADING},
+          },
+        }),
+      });
+      const nextState = reducers(
+        state,
+        actions.graphOpInfoRequested({graph_id: 'g2', op_name: 'Op3'})
+      );
+      expect(nextState.graphs.loadingOps).toEqual({
+        g1: {Op1: DataLoadState.LOADING},
+        g2: {
+          Op2: DataLoadState.LOADING,
+          Op3: DataLoadState.LOADING,
+        },
+      });
+    });
+
+    it('no effect for an already-loading op', () => {
+      const state = createDebuggerState({
+        graphs: createDebuggerGraphsState({
+          loadingOps: {
+            g1: {Op1: DataLoadState.LOADING},
+            g2: {Op2: DataLoadState.LOADING},
+          },
+        }),
+      });
+      const nextState = reducers(
+        state,
+        actions.graphOpInfoRequested({graph_id: 'g2', op_name: 'Op2'})
+      );
+      expect(nextState.graphs.loadingOps).toEqual({
+        g1: {Op1: DataLoadState.LOADING},
+        g2: {Op2: DataLoadState.LOADING},
+      });
+    });
+  });
+
+  describe('graphOpInfoLoaded', () => {
+    it('updates self op, 1 input op and 1 consumer op', () => {
+      const opInfo0 = createTestGraphOpInfo();
+      const state = createDebuggerState({
+        graphs: createDebuggerGraphsState({
+          ops: {
+            g0: {[opInfo0.op_name]: opInfo0},
+          },
+          loadingOps: {
+            g2: {TestOp_1: DataLoadState.LOADING},
+          },
+        }),
+      });
+
+      const opInfo1 = createTestGraphOpInfo({
+        graph_ids: ['g1', 'g2'],
+      });
+      const opInfo2 = createTestGraphOpInfo({
+        op_name: 'TestOp_1',
+        graph_ids: ['g1', 'g2'],
+      });
+      opInfo1.consumers = [
+        [
+          {
+            op_name: opInfo2.op_name,
+            input_slot: 0,
+          },
+        ],
+      ];
+      opInfo2.inputs = [
+        {
+          op_name: opInfo1.op_name,
+          output_slot: 1,
```

Comment:
Sorry, I am missing a bit of context. 

This is the output_slot of the op1, right? How is the op1's output defined? Is it the `consumers`? If so, I only see one output. 

I guess another way to say that is what does this graph look like (visually)?

Reply:
This is a test glitch. Fixed. You're right, op1 only has one output. It didn't cause a test failure before because the affected part was not explicitly asserted on. The assertion was on the whole object. Note the lines that say:

```
     expect(nextState.graphs.ops).toEqual({
        // Verify the old graph op data hasn't changed.
        g0: {[opInfo0.op_name]: opInfo0},
        // 'g2' is the immediately-enclosing graph of the three ops.
        g2: {
          [opInfo1.op_name]: opInfo1,
          [opInfo2.op_name]: opInfo2,
          [opInfo3.op_name]: opInfo3,
        },
      });
```